components/02-elements/button/_button.scss

```diff
@@ -1,121 +1,115 @@
+    background: $button__background;
+    font-family: $button__font-family;
+    font-size: $button__font-size;
+    color: $button__font-color;
+    transition: $button__transition;
+    font-weight: $button__font-weight;
 
-    &:hover {
-        background: $button-hover-background;
-        border-color: $button-hover-border;
+    &:hover,
+    &:focus {
+        background: $button__background-hover;
+        border-color: $button__border-hover;
+        color: $button__font-color-hover;
     }
 
     &--link {
+        display: inline-flex;
         align-items: center;
-        &:hover {
-            text-decoration: none;
-            color: $button-font-color;
-        }
-    }
+        text-decoration: none;
 
-    &--native-link {
-        height: auto;
-        margin: 0;
-        padding: 0;
-        background: none;
-        border: none;
-        color: $link-color;
-        &:hover {
-            background: none;
-            color: $link-hover-color;
-            text-decoration: $link-hover-decoration;
+        &:hover,
+        &:focus {
+            text-decoration: none;
+            color: $button__font-color-hover;
         }
     }
 
     &--secondary {
-        border: $button-secondary-border;
-        color: $button-secondary-font-color;
-        background: $button-secondary-background;
+        border: $button__border--secondary;
+        color: $button__font-color--secondary;
+        background: $button__background--secondary;
 
-        &:hover {
-            border-color: $button-secondary-hover-border;
-            color: $button-secondary-hover-font-color;
-            background: $button-secondary-background;
+        &:hover,
+        &:focus {
+            border-color: $button__border-hover--secondary;
+            color: $button__font-color-hover--secondary;
+            background: $button__background-hover--secondary;
         }
     }
 
     &--icon {
-        padding: $button-icon-padding;
-        border: $button-icon-border;
-        background: $button-icon-background;
+        padding: $button-icon__padding;
+        border: $button-icon__border;
+        background: $button-icon__background;
 
-        &:hover {
-            border-color: $button-icon-hover-border;
-            background: $button-icon-background;
+        &:hover,
+        &:focus {
+            border-color: $button-icon__border-hover;
+            background: $button-icon__background-hover;
             .button__icon {
-                fill:  $button-icon-hover-fill;
+                fill: $button-icon__fill-hover;
             }
         }
-
-        .button__icon {
-            width: $button-icon-size;
-            height: $button-icon-size;
-            fill:  $button-icon-fill;
-        }
-
-        .button__content {
-            margin: $button-icon-content-margin;
-        }
     }
 
     &--fluid {
-        width: $button-fluid-width;
+        width: $button__width--fluid;
+    }
+    .button__icon {
```

Comment:
I've changed the `.button__icon` here  to `&__icon`, I was writing about moving this element here (in the main block) on the begining instead of declaring it inside modifier...  ;)